### PR TITLE
Moving two functions from RSC to RestRequest

### PR DESCRIPTION
**Pull request by vgkholla**
Moving `setDigestAlgorithm()` and `getDigest()` from `ReadableStreamChannel` to `RestRequest`
After examining the use cases, this seems like the right move since not all channels
need to support digests. It is an overhead for many implementations of `ReadableStreamChannel`.

Built and formatted. 

No new tests required. Code was only removed and tests that were removed only tested the production code that was removed.

Addresses #391 

**Primary reviewers: @pnarayanan, @nsivabalan**
**Expected time to review: < 15 mins**

### CODE REVIEW

File: ambry-api/src/main/java/com.github.ambry/rest/RestRequest.java [hunk 1]
```diff
@@ -89,4 +92,33 @@ public void close()
    * @return the {@link RestRequestMetricsTracker} instance attached to this RestRequest.
    */
   public RestRequestMetricsTracker getMetricsTracker();
+
+  /**
```
Comment (vgkholla) [inline ambry-api/src/main/java/com.github.ambry/rest/RestRequest.java]: moved from `ReadableStreamChannel`. All instances of the word "channel" replaced with "data".

File: ambry-commons/src/test/java/com.github.ambry.commons/ByteBufferReadableStreamChannelTest.java [hunk 2]
```diff
@@ -47,10 +44,32 @@
   @Test
   public void commonCaseTest()
       throws Exception {
-    String[] digestAlgorithms = {"", "MD5", "SHA-1", "SHA-256"};
-    for (String digestAlgorithm : digestAlgorithms) {
-      readToAWCTest(digestAlgorithm);
+    ByteBuffer content = ByteBuffer.wrap(fillRandomBytes(new byte[1024]));
```
Comment (vgkholla) [inline ambry-commons/src/test/java/com.github.ambry.commons/ByteBufferReadableStreamChannelTest.java]: This code is not new, it has only been moved - It was in a different function that executed the same code on different digest algorithms. Now that there are no digest algorithms, there was no need for another function.

File: ambry-router/src/test/java/com.github.ambry.router/DeleteManagerTest.java [hunk 3]
```diff
@@ -329,18 +329,18 @@ public void testResponseTimeout()
...
-        CountDownLatch operationCompleteLatch = new CountDownLatch(1);
-        future = router.deleteBlob(blobIdString, new ClientCallback(operationCompleteLatch));
-        do {
-          // increment mock time
-          mockTime.sleep(1000);
-        } while (!operationCompleteLatch.await(10, TimeUnit.MILLISECONDS));
-        assertFailureAndCheckErrorCode(future, expectedError);
-      }
-    });
+          @Override
```
Comment (vgkholla) [inline ambry-router/src/test/java/com.github.ambry.router/DeleteManagerTest.java]: formatting change only

File: ambry-router/src/test/java/com.github.ambry.router/DeleteManagerTest.java [hunk 4]
```diff
@@ -383,14 +383,14 @@ public void testRouterClosedDuringOperation()
...
         RouterErrorCode.RouterClosed, new ErrorCodeChecker() {
-      @Override
-      public void testAndAssert(RouterErrorCode expectedError)
-          throws Exception {
-        future = router.deleteBlob(blobIdString);
-        router.close();
-        assertFailureAndCheckErrorCode(future, expectedError);
-      }
-    });
+          @Override
```
Comment (vgkholla) [inline ambry-router/src/test/java/com.github.ambry.router/DeleteManagerTest.java]: formatting change only

File: ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java [hunk 5]
```diff
@@ -618,26 +618,26 @@ public void onCompletion(final GetBlobResult result, Exception exception) {
...
-                    writableChannel.getNextChunk();
-                    writableChannel.resolveOldestChunk(null);
-                    chunksLeftToRead--;
-                  }
-                  result.getBlobDataChannel().close();
-                } catch (Exception e) {
-                  callbackException.set(e);
-                } finally {
-                  readCompleteLatch.countDown();
+          final ByteBufferAsyncWritableChannel writableChannel = new ByteBufferAsyncWritableChannel();
```
Comment (vgkholla) [inline ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java]: formatting change only

File: ambry-commons/src/test/java/com.github.ambry.commons/ByteBufferReadableStreamChannelTest.java [hunk 6]
```diff
@@ -151,68 +170,6 @@ public void idempotentOperationsTest()
     assertFalse("ByteBufferReadableStreamChannel is not closed", byteBufferReadableStreamChannel.isOpen());
   }
 
-  /**
-   * {@link ByteBufferReadableStreamChannel} has no restrictions on when and what algorithms can be used to generate
-   * the digest. This function tests the correctness of such support.
-   * @throws Exception
-   */
-  @Test
-  public void changingAlgorithmsTest()
```
Comment (nsivabalan) [inline ambry-commons/src/test/java/com.github.ambry.commons/ByteBufferReadableStreamChannelTest.java]: Shouldn't this be moved to NettyRequestTest

Comment (vgkholla) [inline ambry-commons/src/test/java/com.github.ambry.commons/ByteBufferReadableStreamChannelTest.java]: Changing algorithms was a special support of `ByteBufferReadableStreamChannel`. Not all `ReadableStreamChannel` impls (like `NettyRequest`) can support this.

**Pull request merged**